postgres-major-version-upgrades.mdx

```diff
@@ -0,0 +1,114 @@
+published: true
+slug: postgres-major-version-upgrades
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/branch-moving-cover.jpg
+---
+
+One of the challenges of running a Postgres database is upgrading to a new major version. Major version
+upgrades can introduce new features, performance improvements, and bug fixes, but they can also be complex
+and time-consuming.
+
+The most accepted methods for upgrading Postgres to a new major are:
+
+- [pg_upgrade](https://www.postgresql.org/docs/current/pgupgrade.html) for **in-place upgrade**, which involves stopping the server while
+  the upgrade is being performed. Even if this usually happens very fast, it means taking the service down for a few minutes, plus leading
+  with issues in the process during that downtime.
+
+- blue-green deployments, which leverages [logical replication](https://www.postgresql.org/docs/current/logical-replication.html) to create
+  a standby replica (running a newer version) of the primary server. Once replication is done, a switch over is performed, involving
+  cutting writes to the primary, and switching traffic to the replica. Details on this approach later.
+
+Common issues with major Postgres upgrades are:
+
+- **Downtime**: Planned (and unplanned) downtime is not uncommon when a major version upgrade happens, impacting users and business.
+- **Complex orchestration**: Orchestrating a migration is a multi-step process that requires careful planning, with little room
+  for errors.
+- **Adapting clients**: Not only the infra part, in many cases also client apps also need to be aware of the migration and update their
+  database connection details to point to a new host once a migration is done (DNS or Proxies can help here).
+
+For all these reasons, major version upgrades are often postponed and become a major event across multiple engineering teams.
+
+### Moving Database Branches in Xata
+
+Today, we are launching branch moving, which allows Xata users to move a database between two Postgres clusters while it's still being
+accessed. You can use this as a way to move a database from a shared cluster in our free tier to your own dedicated cluster, for better
+performance & isolation.
+
+This feature also allows to perform near zero-downtime major Postgres version upgrades by moving a database between two clusters running
+on different versions.
+
+### How it works
+
+We implemented a blue-green approach in order to offer the ability to move databases. Trying to hide all the complexity behind a simple
+user experience.
+
+In order to understand the process of moving a database, it is useful to explain some key parts of our infrastructure that contribute
+to the process: All Xata databases can be accessed from the same proxy service, that handles authentication and routes traffic to the
+cluster hosting the target database:
+
+![xata clusters and proxy](/images/branch-moving-arch.png)
```

Comment:
@miukimiu do we have or need to have a consistent way to show architecture diagrams? I think we could go live with this, but pinging in case you had a preference.